### PR TITLE
Update zsh packages to 5.5.1

### PR DESCRIPTION
**Pull request by agross**
_Motivation:_ zsh 5.5.1 was released recently
_Linked issues:_
* https://github.com/SynoCommunity/spksrc/issues/2993#issuecomment-342931448 states that zsh is marked as startable, but `spk/zsh/Makefile` [contains `STARTABLE = no`](https://github.com/SynoCommunity/spksrc/blob/master/spk/zsh/Makefile#L17). Unfortunetely I can still `Run` the package using DSM after upgrading to 5.5.1.
  

What also seems to have changed since I submitted PR #2896 is that the [info message](https://github.com/SynoCommunity/spksrc/blob/master/cross/zsh/Makefile#L16-L21) is somehow processed by the build, resulting in these messages at the end of `make` inside e.g. `spk/zsh`. The build succeeds nonetheless.

```
===>  Creating target installation dir of zsh
(cd /spksrc/spk/zsh/work-avoton-6.1/install//usr/local/zsh && tar cpf - `cat /spksrc/spk/zsh/work-avoton-6.1/PLIST | cut -d':' -f2`) | \
  tar xpf - -C /spksrc/spk/zsh/work-avoton-6.1/staging
tar: Building: Cannot stat: No such file or directory
tar: zsh: Cannot stat: No such file or directory
tar: with: Cannot stat: No such file or directory
tar: dynamically: Cannot stat: No such file or directory
tar: linked: Cannot stat: No such file or directory
tar: modules.: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
===>  Stripping binaries and libraries of zsh
...
```

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

**Comment by ymartin59** [comment]
Strange error message... looks like "info" message <<Building zsh with dynamically linked modules.>> from cross/zsh/Makefile has been appended to PLIST ?!?

**Comment by agross** [comment]
Yep, seems like it:

```sh
agross@router ~/spksrc/spk/zsh @master
$ cat work-avoton-6.1/PLIST

Building zsh with dynamically linked modules.
bin:bin/captoinfo
bin:bin/clear
bin:bin/infocmp
bin:bin/infotocap
bin:bin/reset
bin:bin/tabs
bin:bin/tic
bin:bin/toe
bin:bin/tput
bin:bin/tset
bin:bin/zsh
lib:lib/libformw.so.6.1
lib:lib/libmenuw.so.6.1
lib:lib/libncursesw.so.6.1
lib:lib/libpanelw.so.6.1
lnk:lib/libformw.so
lnk:lib/libformw.so.6
lnk:lib/libmenuw.so
lnk:lib/libmenuw.so.6
lnk:lib/libncursesw.so
lnk:lib/libncursesw.so.6
lnk:lib/libpanelw.so
lnk:lib/libpanelw.so.6
shr:share/tabset/std
shr:share/tabset/stdcrt
shr:share/tabset/vt100
shr:share/tabset/vt300
shr:share/terminfo
shr:share/zsh
```

**Comment by Safihre** [comment]
But where? Can you see in which step it happens?

**Comment by agross** [comment]
@Safihre I don't know. In general I find the makefiles hard to follow (I'm not much of a Makefile user). Perhaps zsh/zsh-static is the only package that prints info messages during the build. I'm OK with removing them.

**Comment by Safihre** [comment]
@agross Could you try building again after rebasing on `master`? We have fixed some issues in the main code that caused weird errors during build.

**Comment by agross** [comment]
Still the same output:

```
===>  Creating target installation dir of zsh-static
===>  No PLIST for zsh-static
(cd /spksrc/spk/zsh-static/work-avoton-6.1/install//usr/local/zsh && tar cpf - `cat /spksrc/spk/zsh-static/work-avoton-6.1/PLIST | cut -d':' -f2`) | \
  tar xpf - -C /spksrc/spk/zsh-static/work-avoton-6.1/staging
tar: Building: Cannot stat: No such file or directory
tar: zsh: Cannot stat: No such file or directory
tar: with: Cannot stat: No such file or directory
tar: statically: Cannot stat: No such file or directory
tar: linked: Cannot stat: No such file or directory
tar: modules.: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
===>  Stripping binaries and libraries of zsh-static
```

What's new is the `No PLIST for zsh-static` line.

**Comment by Safihre** [comment]
Did you do a `clean` before?

**Comment by agross** [comment]
Yes

**Comment by ymartin59** [comment]
Published with #3378
